datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -1430,4 +1437,55 @@ public FlowIdentifier updateSalt(SdxCluster sdxCluster) {
+        LOGGER.info("Horizontal scling Data Lake with name {}", name);
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+        final SdxCluster sdxCluster = sdxClusterRepository.findByAccountIdAndClusterNameAndDeletedIsNullAndDetachedIsFalse(accountId, name)
+                .orElseThrow(() -> notFound("SDX cluster", name).get());
+        return sdxReactorFlowManager.triggerHorizontalScaleDataLake(sdxCluster, scaleRequest);
+    }
+
+    public void validateHorizontalScaleRequest(SdxCluster sdxCluster, StackScaleV4Request request) {
+        if (!ENTERPRISE.equals(sdxCluster.getClusterShape())) {
+            throw new BadRequestException(String.format("scaling not supported on: %s.", sdxCluster.getClusterShape().name()));
+        }
+        var hostGroup = request.getGroup();
+        if (!InstanceGroupName.valueOf(hostGroup.toUpperCase(Locale.ROOT)).isScalable()) {
+            throw new BadRequestException(String.format("%s host group cannot scale", hostGroup));
+        }
+        DetailedEnvironmentResponse environmentResponse = environmentClientService.getByName(sdxCluster.getEnvName());
+        if (!EnvironmentStatus.upscalable().contains(environmentResponse.getEnvironmentStatus())) {
+            throw new BadRequestException(String.format("The cluster cannot be scale in the %s status.", environmentResponse.getEnvironmentStatus().name()));
+        }
+        StackV4Response stackResponse = stackV4Endpoint.getByCrn(1L, sdxCluster.getStackCrn(), Set.of());
+        if (isNodeRequestInvalid(stackResponse, request)) {
+            throw new BadRequestException("The requested nodeCount hurt the minimum condition of horizontal sclaing.");
+        }
+    }
+
+    private boolean isNodeRequestInvalid(StackV4Response stack, StackScaleV4Request request) {
+        if (request.getDesiredCount() < 0) {
+            LOGGER.warn("Negative nodeCount was requested.");
+            return true;
+        }
+        var hostgroupMap = stack.getInstanceGroups()
+                .stream()
+                .filter(group -> group.getName().equalsIgnoreCase(request.getGroup()))
+                .collect(toMap(InstanceGroupV4Response::getName, InstanceGroupV4Response::getNodeCount));
+        if (hostgroupMap.isEmpty()) {
+            LOGGER.warn("The reqested hostgroup name not found!");
+            return true;
+        }
+        var instanceGroupName = InstanceGroupName.valueOf(request.getGroup().toUpperCase(Locale.ROOT));
+        if (instanceGroupName.getMinimumNodeCount() > request.getDesiredCount()) {
+            LOGGER.warn("Requested nodeCount is less then the minimum nodeCount.");
+            return true;
+        }
+        if (Math.abs(hostgroupMap.get(instanceGroupName.getName()) - request.getDesiredCount()) < instanceGroupName.getMinimumNodeCount()) {
+            LOGGER.warn("The requested nodeCount will hurt the minimum nodeCount after the scale.");
+            return true;
+        }
```

Comment:
As I understand desired node count after the scaling operation is performed. The check performed above will make sure that scaling operation will not leave the host group less than the minimum required nodes.

This check is not valid. Let's say the core host group has 5 nodes and desired node count provider by the user was 4. According to this condition, the request is not valid. I think this condition should be removed.